main.go

```diff
@@ -215,9 +215,13 @@ func main() {
 	http.Handle("/metrics", prometheus.Handler())
 	http.HandleFunc("/probe",
 		func(w http.ResponseWriter, r *http.Request) {
+			var c Config
+
 			sc.RLock()
-			probeHandler(w, r, sc.C)
+			c = *sc.C
```

Comment:
You should still change this to `c := sc.C`. What you're currently doing is creating a fully copy of the config.